Add dependencies to Docker image & CLI support for CNRelay

TURX opened the pull request:
Fix for issue #383

TURX commented:
You might also want to have a look that it seems that not all music URL requests are tunneled through the CN relay.

pan93412 commented on Dockerfile:
```diff
@@ -1,6 +1,9 @@
 FROM node:lts-alpine
 
 RUN set -ex && mkdir /app
+RUN apk add --no-cache python3 && ln -sf python3 /usr/bin/python
```
The symlink may be not needed if we install those with `apk`.

TURX replied:
If we install Python 3 by ```apk```, ```/bin/sh: python: not found``` will appear if the yt-dlp and youtube-dl bins are downloaded by wget.

TURX replied:
You are right.

TURX replied:
yt-dlp may execute, but youtube-dl (from wget) would have error ```env: can't execute 'python': No such file or directory``` as not like yt-dlp, it requires 'python' instead of 'python3'

TURX replied:
As youtube-dl is provided by apk, I would remove the linking process.

pan93412 commented on Dockerfile:
```diff
@@ -1,6 +1,9 @@
 
 RUN set -ex && mkdir /app
+RUN apk add --no-cache python3 && ln -sf python3 /usr/bin/python
+RUN wget https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -O /usr/local/bin/yt-dlp && chmod a+rx /usr/local/bin/yt-dlp
```
There is a package of it for Alpine Linux, which we used it as our base image: https://pkgs.alpinelinux.org/package/edge/community/x86/youtube-dl

TURX replied:
This works. Thank you for the suggestion.

pan93412 commented on Dockerfile:
```diff
@@ -1,6 +1,9 @@
 RUN set -ex && mkdir /app
+RUN apk add --no-cache python3 && ln -sf python3 /usr/bin/python
+RUN wget https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp -O /usr/local/bin/yt-dlp && chmod a+rx /usr/local/bin/yt-dlp
+RUN wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl && chmod a+rx /usr/local/bin/youtube-dl
```
There is a package of it for Alpine Linux, which we used it as our base image: https://pkgs.alpinelinux.org/package/edge/community/x86_64/yt-dlp

Install it with `apk install --no-cache yt-dlp`.

TURX replied:
There would be an error
```
/app # apk add --no-cache yt-dlp
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/aarch64/APKINDEX.tar.gz
ERROR: unable to select packages:
  yt-dlp (no such package):
    required by: world[yt-dlp]
```

TURX replied:
So I would keep the original method to install yt-dlp.

pan93412 reviewed (changes requested):
Thanks for refactoring this! ❤️ But I found some issues, and could you help me take a look with those? Thanks!

pan93412 commented:
I'll make a E2E test later. Thanks for contribution! :)

1715173329 commented:
Precompiled files should be updated in a separated commit, or just keep them as-is and updated by CI after merge.

TURX commented:
> Precompiled files should be updated in a separated commit, or just keep them as-is and updated by CI after merge.

I have reverted the changes to pre-compiled files.

1715173329 reviewed (approved):
LGTM - Thanks!

pan93412 commented:
I have partially tested it and it passed. I'll squash and merge it now.

<img width="762" alt="截圖 2021-12-16 下午10 00 21" src="https://user-images.githubusercontent.com/28441561/146385914-bdff6d23-75d5-48cc-a0d3-847f4521134e.png">
